adit/dicom_web/utils/wadors_utils.py

```diff
@@ -96,3 +118,181 @@ def callback(ds: Dataset) -> None:
+            logger.error(f"Error during DICOM to NIfTI conversion: {e}")
+            raise
+
+        # Get all files in the output directory using aiofiles instead of sync_to_async(os.listdir)
+        entries = await aiofiles.os.scandir(nifti_output_dir)
+        all_files = [entry.name for entry in entries]
+
+        # Create a mapping of base filenames to their corresponding file paths
+        file_pairs = {}
+        for filename in all_files:
+            base_name, ext = os.path.splitext(filename)
+            if ext == ".json":
+                # For JSON files, store directly
+                file_pairs.setdefault(base_name, {}).update({"json": filename})
+            elif ext == ".gz" and base_name.endswith(".nii"):
+                # For .nii.gz files, strip the .nii part from the base_name
+                actual_base = os.path.splitext(base_name)[0]
+                file_pairs.setdefault(actual_base, {}).update({"nifti": filename})
+            elif ext == ".nii":
+                # For .nii files
+                file_pairs.setdefault(base_name, {}).update({"nifti": filename})
+            elif ext == ".bval":
+                # For .bval files (diffusion b-values)
+                file_pairs.setdefault(base_name, {}).update({"bval": filename})
+            elif ext == ".bvec":
+                # For .bvec files (diffusion b-vectors)
+                file_pairs.setdefault(base_name, {}).update({"bvec": filename})
+
+        # Yield file pairs, JSON first then NIfTI, and finally bval/bvec if they exist
+        for base_name, files in file_pairs.items():
+            # First yield the JSON file if it exists
+            if "json" in files:
+                json_filename = files["json"]
+                json_file_path = os.path.join(nifti_output_dir, json_filename)
+                async with aiofiles.open(json_file_path, "rb") as f:
+                    json_content = await f.read()
+                    yield json_filename, BytesIO(json_content)
+
+            # Then yield the NIfTI file if it exists
+            if "nifti" in files:
+                nifti_filename = files["nifti"]
+                nifti_file_path = os.path.join(nifti_output_dir, nifti_filename)
+                async with aiofiles.open(nifti_file_path, "rb") as f:
+                    nifti_content = await f.read()
+                    yield nifti_filename, BytesIO(nifti_content)
+
+            # Then yield the bval file if it exists (diffusion b-values)
+            if "bval" in files:
+                bval_filename = files["bval"]
+                bval_file_path = os.path.join(nifti_output_dir, bval_filename)
+                async with aiofiles.open(bval_file_path, "rb") as f:
+                    bval_content = await f.read()
+                    yield bval_filename, BytesIO(bval_content)
+
+            # Finally yield the bvec file if it exists (diffusion b-vectors)
+            if "bvec" in files:
```

Comment:
The if clauses look very redundant, why not a for loop like this:
for file_type in ["json", "nifti", ...]:
  filename = files[file_type]
  ...